engine/api/src/main/scala/pl/touk/nussknacker/engine/api/context/UsageClosure.scala

```diff
@@ -0,0 +1,23 @@
+package pl.touk.nussknacker.engine.api.context
+
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedClass, TypingResult}
+
+sealed trait UsageClosure
+
+case class VariableName(name: String) extends UsageClosure
+
+// TODO: should be Expression in values?
+case class BranchExpressions(expresionByBranchId: Map[String, ValidationContext => TypedClass]) extends UsageClosure
+
+case class CompileTimeEvaluatedParameter[T](name: String, value: T) extends UsageClosure
```

Comment:
I think here there should be other options, not all parameters can be evaluated in compile time: 
- TypingResult
- Expression
- ValidationContext => TypingResult